### PR TITLE
Bugfix/two czi#15

### PR DESCRIPTION
**Pull request by heeler**
The fixes implemented here address the bug triggered by closing the file pointer. 

The fix involved refactoring the code to make it cleaner as well. (No FileHandler class anymore)

### CODE REVIEW

File: pylibczi/tests/test_czi_file.py [hunk 1]
```diff
@@ -43,6 +43,16 @@ def test_is_mosaic(data_dir, fname, expected):
...
 
 
+@pytest.mark.parametrize("fname, expected", [
+    ('s_1_t_1_c_1_z_1.czi', (False)),
+    ('s_3_t_1_c_3_z_5.czi', (False)),
+])
+def test_destructor(data_dir, fname, expected):
+    czi = CziFile(str(data_dir / fname))
+    czi = CziFile(str(data_dir / fname))
+    #this should segfault on windows
```
Comment (toloudis) [inline pylibczi/tests/test_czi_file.py]: should segfault?

File: setup.py [hunk 2]
```diff
@@ -119,8 +121,8 @@ def build_extension(self, ext):
 
         if platform.system() == "Windows":
             cmake_args += ['-DCMAKE_LIBRARY_OUTPUT_DIRECTORY_{}={}'.format(cfg.upper(), extdir)]
-            if sys.maxsize > 2**32:
-                cmake_args += ['-A', 'x64']
+            # if sys.maxsize > 2**32:
+            #    cmake_args += ['-A', 'x64']
```
Comment (toloudis) [inline setup.py]: can remove these lines

File: test.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import pylibczi
```
Comment (toloudis) [inline test.py]: should this file still be here?

File: pylibczi/tests/test_czi_file.py (hunk 1 above)
Comment (heeler) [reply pylibczi/tests/test_czi_file.py]: I confirmed this does segfault (without the fix) added pytest-xdist and changed workflow to run pytest -n1 ...  This causes the segfault to be captured and the remaining tests run but now the github actions should work.

**Comment by heeler** [comment]
I closed this after addressing dan's comments. Then the PR's are clear as the diff is against master (ie master + twoCzi). If you have comments please add them to the other PR for subblock metadata

**Pull request merged**